File: 프로그래머스/lv1/12933. 정수 내림차순으로 배치하기/정수 내림차순으로 배치하기.java

import java.util.Arrays;
import java.util.Collections;
class Solution {
    public long solution(long n) {
        
        String str = String.valueOf(n);
        String[] arr = str.split("");
        
        Arrays.sort(arr, Collections.reverseOrder());
        
        String result = "";
        for(int i=0; i<arr.length; i++) {
            result += arr[i];
        }
        
        return Long.parseLong(result);
    }
}
/*
    접근법 :
    1. 매개변수 n을 문자열로 변환 -> String.valueOf()
    2. 변환된 문자열을 문자 하나하나 배열로 만들기 -> .split("")
    3. 배열의 원소들을 내림차순 정렬 -> Arrays.sort(arr, Collections.reverseOrder());
    4. 배열을 순회하며 빈 문자열에 넣어준다.
    5. 완성된 문자열을 long 타입으로 변환해 반환

    
    다른 접근법 :
    import java.util.*
    class Solution {
        public long solution(long n) {

            String[] str = String.valueOf(n).split("");
            Arrays.sort(str);

            StringBuilder sb = new StringBuilder();
            for(String result : str) {
                sb.append(result);
            }

            return Long.parseLong(sb.reverse().toString());
        }
    }
*/

File: 프로그래머스/lv1/12933. 정수 내림차순으로 배치하기/정수 내림차순으로 배치하기.js
function solution(n) {
    const result = String(n).split("").map( (a) => {
        return Number(a)
    }).sort( (a,b) => {
        return b-a
    }).join("");
    return Number(result);
}
/*
    접근법 :
    1. 정수로 들어온 매개변수를 문자열로 변환 -> String()
    2. 각각의 문자를 배열의 원소로 만들기 -> split("")
    3. 문자열 원소들을 정수로 변환 -> map()
    4. 정수로 변환된 원소들을 내림차순 정렬 -> sort()
    5. 정수로 이루어진 배열을 하나의 문자열로 만들기 -> join("")
    6. 문자열인 정수를 정수로 반환 -> Number()


    다른 접근법 :
    처음에 접근한 작성법에서 조금 간결하게 줄여보았다.

    function solution(n) {
        return Number(String(n).split("").sort().reverse().join(""));       
    }
*/